core/templates/pages/story-editor-page/services/story-editor-state.service.ts

```diff
@@ -35,28 +36,40 @@ export class StoryEditorStateService {
   constructor(
     private alertsService: AlertsService,
     private editableStoryBackendApiService: EditableStoryBackendApiService,
+    private loaderService: LoaderService,
     private storyObjectFactory: StoryObjectFactory,
     private undoRedoService: UndoRedoService) {}
 
-  _story: Story = this.storyObjectFactory.createInterstitialStory();
   _storyIsInitialized: boolean = false;
   _storyIsLoading: boolean = false;
   _storyIsBeingSaved: boolean = false;
-  _topicName: string = null;
   _storyIsPublished: boolean = false;
   _skillSummaries: SkillSummaryBackendDict[] = [];
   _expIdsChanged: boolean = false;
   _storyWithUrlFragmentExists: boolean = false;
-  _classroomUrlFragment: string = null;
-  _topicUrlFragment: string = null;
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  _story!: Story;
+  _topicName!: string;
+  _classroomUrlFragment!: string;
+  _topicUrlFragment!: string;
 
   _storyInitializedEventEmitter = new EventEmitter();
   _storyReinitializedEventEmitter = new EventEmitter();
   _viewStoryNodeEditorEventEmitter = new EventEmitter();
   _recalculateAvailableNodesEventEmitter = new EventEmitter();
 
   private _setStory(story: Story): void {
-    this._story.copyFromStory(story);
+    if (!this._story) {
+      // The Story is set directly for the first load.
+      this._story = story;
+    } else {
+      // After first initialization, the story object will be retained for
+      // the lifetime of the editor and on every data reload or update, the new
+      // contents will be copied into the same retained object.
+      this._story.copyFromStory(story);
```

Comment:
Do both of these statements(at `66` and `71`) achieve the same thing?

Reply:
Yes, but there is one difference in their usage. We cannot use copyFromStory on story1 to copy the value of story2 to it if story1 is not defined (initialized). But when we call loadStory for the first time, story1 will always be undefined, so we will assign the value of story2 directly to it.

Reply:
I am not sure, but I guess we need to do a deep copy of the story after we get the defined value of the story the first time. I used a similar pattern that we have used for [skill-editor-state-service](https://github.com/oppia/oppia/blob/324782bdff60829240a21d42c5f81f06e1413c9f/core/templates/pages/skill-editor-page/services/skill-editor-state.service.ts#L80).